### PR TITLE
NNS1-3485: styling for the Report page

### PR DESCRIPTION
**Pull request by yhabib**
# Motivation

Styles for the report page.

# Changes

- Styles for the Reporting page.

# Tests

- Not necessary.

# Todos

- [ ] Add entry to changelog (if necessary).
Not necessary

### CODE REVIEW

File: frontend/src/lib/routes/Reporting.svelte [hunk 1]
```diff
@@ -17,6 +23,34 @@
       500
     )
   );
+
+  let identity: Identity | null | undefined;
+  let neurons: NeuronInfo[] = [];
+
+  $: identity = $authStore.identity;
+
+  const loadNeurons = async (id: typeof identity) => {
```
Comment (dskloetd) [inline frontend/src/lib/routes/Reporting.svelte:32]: Why not `Identity` instead of `typeof identity`?

Comment (dskloetd) [inline frontend/src/lib/routes/Reporting.svelte:32]: `id` can mean lots of things (neuron ID, canister ID, etc.). Let's stick with `identity`.

File: frontend/src/lib/routes/Reporting.svelte [hunk 2]
```diff
@@ -17,6 +23,34 @@
...
+    if (!id) return;
+
+    try {
+      const data = await queryNeurons({
+        certified: true,
+        identity: id,
+        includeEmptyNeurons: true,
+      });
+
+      neurons = sortNeuronsByStake(data);
```
Comment (dskloetd) [inline frontend/src/lib/routes/Reporting.svelte:42]: Isn't it better to sort them by creation date?
That way you guarantee a consistent order between exports.
Stake can change over time but creation date will never change.

File: frontend/src/lib/routes/Reporting.svelte [hunk 3]
```diff
@@ -17,6 +23,34 @@
...
+    } catch (err) {
+      console.error("Failed to load neurons:", err);
+      toastsError({
+        labelKey: "reporting.fetching_neurons_error",
+      });
+    }
+  };
+
+  $: if (identity) {
```
Comment (dskloetd) [inline frontend/src/lib/routes/Reporting.svelte:51]: Use `nonNullish`.

File: frontend/src/lib/routes/Reporting.svelte [hunk 4]
```diff
@@ -17,6 +23,34 @@
...
+    loadNeurons(identity);
```
Comment (dskloetd) [inline frontend/src/lib/routes/Reporting.svelte:52]: Why not wait for a button press before downloading the neurons?

File: frontend/src/tests/lib/routes/Reporting.spec.ts [hunk 5]
```diff
@@ -30,4 +47,23 @@ describe("Reporting", () => {
         title: en.navigation.reporting,
       }));
   });
+
+  it("should reload neurons when identity changes", async () => {
```
Comment (dskloetd) [inline frontend/src/tests/lib/routes/Reporting.spec.ts:51]: How does this test change the identity?

File: frontend/src/tests/lib/routes/Reporting.spec.ts [hunk 6]
```diff
@@ -30,4 +47,23 @@ describe("Reporting", () => {
...
+    // Wait for initial load
+    renderComponent();
+    await runResolvedPromises();
+
+    expect(spyQueryNeurons).toHaveBeenCalledTimes(1);
+  });
+
+  it("should show a toast error if queryNeurons fails", async () => {
```
Comment (dskloetd) [inline frontend/src/tests/lib/routes/Reporting.spec.ts:59]: Also test that there is no toast if there is no error.

File: frontend/src/lib/routes/Reporting.svelte (hunk 2 above)
Comment (yhabib) [reply frontend/src/lib/routes/Reporting.svelte:44]: This was a requirement from Andrew. As some users will get a lot of empty neurons, they want to push them down.
I'll note this comment and bring it to him for the next iteration.

File: frontend/src/lib/routes/Reporting.svelte (hunk 4 above)
Comment (yhabib) [reply frontend/src/lib/routes/Reporting.svelte:54]: I was thinking of disabling the buttons based on certain conditions one being the neurons, though we don't have a proper message to show in the UI 🤔

Comment (yhabib) [reply frontend/src/lib/routes/Reporting.svelte:54]: If it’s alright with you, I’ll follow up with Andrew tomorrow regarding the introduction of the other components to the page.

File: frontend/src/tests/lib/routes/Reporting.spec.ts
Comment (dskloetd) [inline frontend/src/tests/lib/routes/Reporting.spec.ts:1]: Changes in this seem unrelated to the PR?

**Review by dskloetd — APPROVED** [review]
